src/borg/archiver.py

```diff
@@ -364,8 +364,8 @@ def do_key_export(self, args, repository):
             manager.export_paperkey(args.path)
         else:
             if not args.path:
-                self.print_error("output file to export key to expected")
-                return EXIT_ERROR
+                print(manager.get_keyfile_data())
+                return EXIT_SUCCESS
```

Comment:
we are getting closer, but still, we are not there yet...

now we have the strange situation that the code in 367/368 implements just another behaviour (based on absence of a path, but it does not care about whether qr.args is given or not) in parallel to the behaviour in 370..373 (which is different based on whether args.qr is given or not).

i really think the block in 366..368 needs to get completely removed.

and as a replacement, `KeyManager.export` and `.export_qr` need to be made tolerant of getting path=None or path='-' and just use sys.stdout in that case.

Reply:
so, `store_keyfile` and `export_qr` could just call `dash_open` instead of plain `open`.

with that, everything left to do is to use `'-'` (the dash) in case args.path is None (which can be also handled right at the beginning of `export` and `export_qr`).

export_paperkey should then also get a little refactored and the `if path` block at its end being replaced by `dash_open` also.

Reply:
with that, we get:
- consistent behaviour no matter which options are given
- with no path or with path == '-', it uses stdout
- less code, reusing the dash_open code